public/app/plugins/panel/table2/TablePanel.tsx

```diff
@@ -30,20 +32,71 @@ export class TablePanel extends Component<Props> {
     });
   };
 
+  onChangeTableSelection = (val: SelectableValue<number>) => {
+    this.props.onOptionsChange({
+      ...this.props.options,
+      frameIndex: val.value || 0,
+    });
+
+    // Force a redraw -- but no need to re-query
+    this.forceUpdate();
+  };
+
+  renderTable(frame: DataFrame, width: number, height: number) {
+    const {
+      options: { showHeader, resizable },
+    } = this.props;
+    return <Table height={height} width={width} data={frame} noHeader={!showHeader} resizable={resizable} />;
+  }
+
   render() {
     const {
       data,
       height,
       width,
-      options: { showHeader, resizable },
+      options: { frameIndex },
     } = this.props;
 
-    if (data.series.length < 1) {
+    const count = data.series?.length;
+    if (!count || count < 1) {
       return <div>No Table Data...</div>;
     }
-
-    return (
-      <Table height={height - 16} width={width} data={data.series[0]} noHeader={!showHeader} resizable={resizable} />
-    );
+    if (count > 1) {
+      const index = frameIndex > 0 && frameIndex < count ? frameIndex : 0;
+      const names = data.series.map((frame, index) => {
+        return {
+          label: `${frame.name ?? 'Series'}`,
+          value: index,
+        };
+      });
+      return (
+        <div className={tableStyles.wrapper} style={{ height }}>
+          <div className={tableStyles.table}>
+            <AutoSizer>
+              {({ width, height }) => {
+                if (height < 5) {
+                  return null;
+                }
+                height = height - 5;
+                return <div style={{ width, height }}>{this.renderTable(data.series[index], width, height)}</div>;
+              }}
+            </AutoSizer>
```

Comment:
Not a fan of another auto sizer here (they are a bit expensive, slow down initial rendering), we can easily calculate the height of the select and deduct it from the panel height

Reply:
theme.spacing.formInputHeight  (plus whatever padding/margin)